crates/phactory/api/src/crypto.rs

```diff
@@ -178,28 +215,39 @@ impl CertificateBody {
         msg_type: MessageType,
         sig_type: SignatureType,
         signature: &[u8],
-    ) -> Result<Vec<u8>, SignatureVerifyError> {
-        match sig_type {
+    ) -> Result<AccountId32, SignatureVerifyError> {
+        let signer = match sig_type {
             SignatureType::Ed25519 => {
-                recover::<sp_core::ed25519::Pair>(&self.pubkey, signature, msg)
+                recover::<sp_core::ed25519::Pair>(&self.pubkey, signature, msg)?.into()
             }
             SignatureType::Sr25519 => {
-                recover::<sp_core::sr25519::Pair>(&self.pubkey, signature, msg)
+                recover::<sp_core::sr25519::Pair>(&self.pubkey, signature, msg)?.into()
             }
-            SignatureType::Ecdsa => recover::<sp_core::ecdsa::Pair>(&self.pubkey, signature, msg),
+            SignatureType::Ecdsa => account_id_from_evm_pubkey(recover::<sp_core::ecdsa::Pair>(
```

Comment:
Should we leave the original Substrate ECDSA implementation, and introduce EvmEcdsa as an alternative?

Reply:
Make sense. Done.